KS-4-5 задача 4

VVShchegolev opened the pull request.

Commit:
KS-4-5 задача 4

ievetrovdev commented on src/main/kotlin/Lesson_4_Ex_4/Lesson_4_Ex_4.kt:
```diff
@@ -0,0 +1,16 @@
+
+    val trainingDay = 5
+
+    // Определяем, какие упражнения выполнять (через день: руки/пресс на нечетные дни, ноги/спина на четные)
```
Комментарии в коде принято оставлять только для объяснения какой-то сложной бизнес-логики. За всё остальное в коде должны говорить корректные названия переменных, функций, классов и понятная структура кода. Давай удалим комментарии.

ievetrovdev commented on src/main/kotlin/Lesson_4_Ex_4/Lesson_4_Ex_4.kt:
```diff
@@ -0,0 +1,16 @@
+    val trainingDay = 5
+
+    // Определяем, какие упражнения выполнять (через день: руки/пресс на нечетные дни, ноги/спина на четные)
+    val isArmsAndAbsDay = trainingDay % 2 == 1 // Нечетные дни: руки и пресс
```
Верный ход, но давай оптимизируем. Предлагаю четность числа сохранять в Boolean "флаг" с более универсальным названием, например isEven. И применять его уже при выводе “как есть” или с отрицанием (инвертируя значение). Не усложняя код лишними if. И более принято находить именно четность) сравнивая с 0, а не с 1.

VVShchegolev committed:
Update KS_4_4 Задача 4

Что было исправлено:
Были удалены комментарии;
Изменил название переменной "isArmsAndAbsDay" на "isEven";
Применил эту переменную при выводе "Как есть" или с отрицанием, вместо if;
Заменил 0 и 2 на константы, чтобы не было магических чисел.

ievetrovdev reviewed (approved):
Отлично
